rfcs/20200827-keras-cv-scoping-design.md

```diff
@@ -0,0 +1,115 @@
+- Boundaries between `keras-cv` and [TensorFlow model garden](https://github.com/tensorflow/models)
+- Boundaries between `keras-cv` and [tf.keras.applications](https://keras.io/api/applications/)
+
+## Motivation
+
+Computer vision (CV) is a major application area for our users.
+Keras on its own provides good support for image classification tasks, in particular via `tf.keras.applications`.
+However, a Keras-native modeling solutions for more advanced tasks,
+such as object detection, instance segmentation, etc., is still lacking.
+
+As a result, the open-source community has rolled out many different solutions for these use cases,
+made available via PyPI and GitHub. These third-party solutions are not always kept up to date, and
+many still rely on the legacy multi-backend Keras. They also raise the issue of API standardization.
+
+To fix this, we want machine learning engineers to have access to a standard Keras-native,
+optimized, and well-tested set of components to build their advanced computer vision models.
+
+This provides key user benefits:
+
+- The package would be first-party and thus always up to date with modern best practices.
+- High code quality and testing standards and strict quality control: same level of trust as core Keras
+- A shared API standard across the community
+- Ability for the open-source community to build more advanced solutions *on top* of this package instead of reinventing it
+
+## Design Proposal
+
+`keras-cv` will provide components that cover the following areas:
+
+- Object Detection tasks.
+- Instance Segmentation tasks.
+- Semantic Segmentation tasks.
+- Keypoint Detection tasks.
+- Video Classification tasks.
+- Object Tracking tasks.
+
+Specifically, for Object Detection tasks, `keras-cv` will include most anchor-based modules:
+
+- Common objects such as anchor generator, box matcher.
+- Keras layer components such as ROI generator, NMS postprocessor.
+- Keras backbone components that fills the gap from keras-applications.
+- Keras losses and metrics, such as Focal loss and coco metrics.
+- Data loader and preprocessing for different dataset, such as COCO.
+
+For Semantic Segmentation tasks, `keras-cv` will include:
+
+- Keras head components such as Atrous Spatial Pyramid Pooling (ASPP).
+
+## Success criteria for `keras-cv`
+
+- Cover all modeling tasks listed above
+- Easy-to-use API
+- Models run on CPU/GPU/TPU seamlessly
+- State of the art performance
+- Models can be readily deployed to production
+
+## Boundaries between keras-cv and keras-applications
+
+- keras-applications will be improved to include basic building blocks such as mobilenet bottleneck, that
+ include feature maps
+- keras-cv will depend on keras-applications for importing backbones.
+
+## Boundaries between keras-cv and Tensorflow Addons
+
+- Highly experimental modeling, layers, losses, etc, live in addons.
+- Components from addons will graduate to Model Garden, given it incurs more usage,
```

Comment:
Similar question regarding Addons graduation as in the NLP RFC

Reply:
Same response as NLP scoping doc.